### PR TITLE
[charts] Fix Barchart with empty dataset throwing an error

### LINKED ISSUE #12462: [Charts] Barchart with empty dataset should render an empty chart
### Steps to reproduce

Link to live example: https://stackblitz.com/edit/react-vvrvzj?file=Demo.tsx

Steps:
1. open demo
2.
3.

### Current behavior

```
Error in /turbo_modules/@mui/x-charts@6.19.5/BarChart/extremums.js (22:36)
undefined is not iterable (cannot read property Symbol(Symbol.iterator))
```

### Expected behavior

An empty chart is shown, as if `series={[]}`

### Context

_No response_

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: empty barchart

### PR DESCRIPTION
**Pull request by JCQuintas**
We just use optional chain and nullchecks as fallbacks.

@alexfauquette Should we add tests for these functions?

Fixes #12462

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

**Comment by alexfauquette** [comment]
I agree it fixes the issue, but I'm wondering why do we have this issue with bar chart and not with line chart? They are super close, so they should have the same behavior with empty dataset.

For the fallback on `[null, null]` I'm ok, it's the same behavior, but for the use of optional chain I'm wondering if we are missing something since I don't see them used in the line chart

**Comment by JCQuintas** [comment]
@alexfauquette In the Line chart it is divided into two functions and we check for 

https://github.com/JCQuintas/mui-x/blob/3cee37ded0a82ab1b2854fa6f2a430c29c32f090/packages/x-charts/src/LineChart/extremums.ts#L17-L19

``` js 
if (stackedData.length === 0) {
    return [null, null];
}
```

**Comment by JCQuintas** [comment]
Added tests, odd behaviour is that vertical layout returns `[Infinity, -Infinity]` while horizontal is returning `[null, null]`. Doesn't seem to affect anything though

**Comment by alexfauquette** [comment]
Not that surprising. It comes from `Math.min()` returning `Infinity`.

```
const getBaseExtremum: ExtremumGetter<'bar'> = (params) => {
  const { axis } = params;

  const minX = Math.min(...(axis.data ?? []));
  const maxX = Math.max(...(axis.data ?? []));
  return [minX, maxX];
};
```

What is more surprising is that those extremes are computed. For `"band"` and `"point"` scales, the notion of extreme values does not make sense because they contain categories. So they do not have an notion of order.

In cartesian context, those scales got an [early return](https://github.com/mui/mui-x/blob/master/packages/x-charts/src/context/CartesianContextProvider.tsx/#L218-L221) before the min/max values are used. So they could return wherever they want.

Maybe the correct behavior would be that in `vertical` it always returns `[null, null]` but since this value is never used, I don't have strong opinion about it

File: packages/x-charts/src/BarChart/extremums.test.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    },
+    axis: {
+      id: 'id',
+      data,
+    },
+    isDefaultAxis: true,
+  };
+};
+
+describe('BarChart - extremums', () => {
```
Comment (oliviertassinari) [inline packages/x-charts/src/BarChart/extremums.test.ts:33]: Time will tell if this test isn't too low level. Testing public APIs tends to scale better.

**Pull request merged**